test/hotspot/jtreg/containers/docker/TestMemoryAwareness.java

```diff
@@ -153,18 +153,24 @@ private static void testOperatingSystemMXBeanAwareness(String memoryAllocation,
         out.shouldHaveExitValue(0)
            .shouldContain("Checking OperatingSystemMXBean")
            .shouldContain("OperatingSystemMXBean.getTotalPhysicalMemorySize: " + expectedMemory)
-           .shouldMatch("OperatingSystemMXBean\\.getFreePhysicalMemorySize: [1-9][0-9]+")
            .shouldContain("OperatingSystemMXBean.getTotalMemorySize: " + expectedMemory)
            .shouldMatch("OperatingSystemMXBean\\.getFreeMemorySize: [1-9][0-9]+")
-           .shouldMatch("OperatingSystemMXBean\\.getFreeSwapSpaceSize: [1-9][0-9]+");
-        // in case of warnings like : "Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap."
-        // the getTotalSwapSpaceSize returns the system values as the container setup isn't supported in that case.
+           .shouldMatch("OperatingSystemMXBean\\.getFreePhysicalMemorySize: [1-9][0-9]+");
+
+        // in case of warnings like : "Your kernel does not support swap limit capabilities
+        // or the cgroup is not mounted. Memory limited without swap."
+        // the getTotalSwapSpaceSize does not return the expected result and
+        // getFreeSwapSpaceSize returns 0
         try {
             out.shouldContain("OperatingSystemMXBean.getTotalSwapSpaceSize: " + expectedSwap);
         } catch(RuntimeException ex) {
-            out.shouldMatch("OperatingSystemMXBean.getTotalSwapSpaceSize: [1-9][0-9]+");
-            out.shouldContain("Metrics.getMemoryLimit() == " + expectedMemory);
-            out.shouldContain("Metrics.getMemoryAndSwapLimit() == -1");
+            out.shouldMatch("OperatingSystemMXBean.getTotalSwapSpaceSize: -?([0-9]+)");
```

Comment:
Is the optional `-` intentional? My understanding is that it should never be negative, should it not?

Reply:
The optional - is intential in case UNLIMITED is return (-1).

Reply:
Hmm, but the OperatingSystemMXBean impl falls back to returning the system (host) values if the container limits are unlimited. So the internal Metrics value of -1 for UNLIMITED should never be seen. Has this been actually seen in some tests somewhere?

Reply:
https://github.com/openjdk/jdk/blob/a75edc29c6ce41116cc99530aa1710efb62c6d5a/src/jdk.management/unix/classes/com/sun/management/internal/OperatingSystemImpl.java#L63

Reply:
The -1 has not been seen.  I removed the option '-' from the pattern.  Please see the changes in the latest commit.  Thanks!

Reply:
I thought we were seeing -1 but perhaps this was a previous result before we decided to change Swap+Memory to report Memory instead of failure.  I'm going to rerun the tests with and without swap enabled to verify.

Reply:
All container tests pass with and without swap enabled after removing the "-".